TarSCM/archive.py

```diff
@@ -104,6 +104,15 @@ def create_archive(self, scm_object, **kwargs):
 
         if commit:
             metafile.write("commit: " + commit + "\n")
+            names = {"%ci": "commit_date: ",
+                     "%h": "short_commit_hash: ",
+                     "@TAG_OFFSET@": "commits_since_last_tag: "}
+
+            for field in ("%h", "%ci", "@TAG_OFFSET@"):
+                fake_args = args.copy()
+                fake_args["versionformat"] = field
+                value = str(scm_object.detect_version(fake_args))
+                metafile.write(names[field] + value + "\n")
 
```

Comment:
looks very specific to git to me. Maybe this should be handled in a SCM specific method and TarSCM.scm.base should have a generic method which simply 'pass'. Otherwise I guess it could break e.g. svn/bzr/...

Reply:
Yeah I agree, git-specific stuff should remain in the `Git` class.